pkg/services/object/get/assembly_v2.go

```diff
@@ -57,6 +57,8 @@ func (exec *execCtx) processV2Link(linkID oid.ID) bool {
 	linkObj, ok := exec.getChild(linkID, nil, true)
 	if !ok {
 		exec.log.Debug("failed to read link object")
+		exec.status = statusNotFound
+		exec.err = apistatus.ErrObjectNotFound
 		return false
```

Comment:
But why `processV2Split()` doesn't handle this `false` correctly already? It's supposed to handle link object problems, `true` is returned when the status from `processV2Link` is final and `false` is returned when we should fallback to chain processing.